imapclient/id.go

```diff
@@ -0,0 +1,167 @@
+package imapclient
+
+import (
+	"fmt"
+
+	"github.com/emersion/go-imap/v2"
+	"github.com/emersion/go-imap/v2/internal/imapwire"
+)
+
+// Id sends an ID command.
+// ID command is introduced in RFC 2971.
+// An example ID command:
+// ID ("name" "go-imap" "version" "1.0" "os" "Linux" "os-version" "7.9.4" "vendor" "Yahoo")
+func (c *Client) Id(idData *imap.IdData) *IdCommand {
+	cmd := &IdCommand{}
+	enc := c.beginCommand("ID", cmd)
+
+	if idData == nil {
+		enc.SP().NIL()
+		enc.end()
+		return cmd
+	}
+
+	enc.SP().Special('(')
+	isFirstKey := true
+	if idData.Name != "" {
+		addIdKeyValue(enc, isFirstKey, "name", idData.Name)
+		isFirstKey = false
```

Comment:
Maybe the address of `isFirstKey` could be passed to `addIdKeyValue` to avoid repetition and always set the value to `true` at the end of the call. Alternatively, build a `[]string` of key-values and encode them in one loop.